rust/kcl-lib/src/execution/mod.rs

```diff
@@ -1446,6 +1452,44 @@ pub(crate) struct ExecTestResults {
     exec_state: ExecState,
 }
 
+/// There are several places where we want to traverse a KCL program or find a symbol in it,
+/// but because KCL modules can import each other, we need to traverse multiple programs.
+/// This stores multiple programs, keyed by their module ID for quick access.
+#[cfg(feature = "artifact-graph")]
+pub struct ProgramLookup {
+    programs: IndexMap<ModuleId, crate::parsing::ast::types::Node<crate::parsing::ast::types::Program>>,
+}
+
+#[cfg(feature = "artifact-graph")]
+impl ProgramLookup {
+    // TODO: Could this store a reference to KCL programs instead of owning them?
+    // i.e. take &state::ModuleInfoMap instead?
+    pub fn new(
+        current: crate::parsing::ast::types::Node<crate::parsing::ast::types::Program>,
+        module_infos: state::ModuleInfoMap,
+    ) -> Self {
+        let mut programs = IndexMap::with_capacity(module_infos.len());
+        for (id, info) in module_infos {
+            #[cfg(target_arch = "wasm32")]
+            web_sys::console::log_1(&format!("ProjectProgramLookup module {}: {:?}", id.as_usize(), info.path).into());
```

Comment:
Is all the console logging just for debugging? I would probably delete them.

Reply:
Oops, yes it was for debugging. I'll delete them all.